library/src/main/java/com/chuckerteam/chucker/api/BodyDecoder.kt

```diff
@@ -0,0 +1,29 @@
+package com.chuckerteam.chucker.api
+
+import okhttp3.Request
+import okhttp3.Response
+import okio.ByteString
+import okio.IOException
+
+/**
+ * Decodes HTTP request and response bodies to human–readable texts.
+ */
+public interface BodyDecoder {
+    /**
+     * Returns a text representation of [body] that will be displayed in Chucker UI transaction,
+     * or `null` if [request] cannot be handled by this decoder. [Body][body] is no longer than
+     * [max content length][ChuckerInterceptor.Builder.maxContentLength] and is guaranteed to be
+     * gunzipped even if [request] has gzip header.
+     */
+    @Throws(IOException::class)
+    public fun decodeRequest(request: Request, body: ByteString): String?
```

Comment:
nit: This is more an API design question.

What we're modelling here is an operation that could have 3 outcomes: success (`String`), failure (`IOException`), skipped (`null`).

Have you considered different approaches (e.g. `sealed classes` or the Kotlin `Result` type)? I'm not saying that the approach here is wrong, but I was wondering if we have some benefit in modelling it differently.

Reply:
Agree with this nitpick that it might be a good idea to model the result to leave less ambiguity. 
For example, we could have something like `Failure` instead of null in case of unsupported content.

Reply:
Yes, I considered it I don't think it is the right choice. I don't see how it helps us in the processing pipeline. We are interested only in the first, successfully decoded body.

`IOException` does not model a type of respons. Methods are annotated with it because tools like adapters generally can throw this type of exception so it removes form users a burden of needing to deal with it. And in Kotlin it is easy to forget since exceptions are not checked. Also, it allows us to log exceptions for the user so they have feedback information when something goes wrong. I would be either for keeping this part as is or to remove `@Throws` and move the responsibility to the users.

Returning `null` instead of some type is a nod towards libraries like Retrofit or Moshi, where factories return it when they can't handle input. I don't mind modelling it with a sealed class but I don't know if it brings anything to the table. If you feel that it would be helpful I'll change it.

Reply:
Thanks for your insights 👍 I think we're fine with the current modelling for the reasons you mentioned + is the one that has the smaller API surface

Reply:
This thread started with `nit` annotation, so let's leave it as it is :)